### ISSUE TITLE
Tables when using mdformat-gfm fail to parse some scenarios, however, appear valid on github

### ISSUE THREAD

**Issue by butler54** [executablebooks/mdformat#281]
### Describe the problem

I ran into an issue when using mdformat-gfm which triggers the generic:
```
Error: Could not format "test_tables.md".

The formatted Markdown renders to different HTML than the input Markdown. This
is likely a bug in mdformat. Please create an issue report here, including the
input Markdown: https://github.com/executablebooks/mdformat/issues
```

 I've distilled this down to a set of known reproducers below. I've also isolated that these issues *do not* occur when just using mdformat-tables. Only when using mdformat-gfm (plus the pre-req mdformat tables).

# Rendered table:

| Description | Currently parsing? | Test column | 
| -----------  | ------------------ | ------------ |
| Adding italics | NO | _Italicised_ |
| Adding a domain with a port | NO | test.com:443 | 
| Adding a domain w/o a port | YES | test.com | 
| Adding something like a domain with a colon | YES | testcom:443 | 
| Adding a domain with port and protocol | YES | https://test.com:443 |
| Adding code blocks | YES | `whatever.com:443` | 

# actual markdown

```markdown
| Description | Currently parsing? | Test column | 
| -----------  | ------------------ | ------------ |
| Adding italics | NO | _Italicised_ |
| Adding a domain with a port | NO | test.com:443 | 
| Adding a domain w/o a port | YES | test.com | 
| Adding something like a domain with a colon | YES | testcom:443 | 
| Adding a domain with port and protocol | YES | https://test.com:443 |
| Adding code blocks | YES | `whatever.com:443` | 
```

### Steps to reproduce

1. Install mdformat-gfm
2. Run mdformat on the table above.
3. Remove individual rows to isolate particular issue of interest.

### The version of Python you're using

3.9.7

### Your operating system

OSX latest

### Versions of your packages

```
(mdformat) mdformat_tests % pip freeze
attrs==21.2.0
importlib-metadata==4.8.1
linkify-it-py==1.0.2
markdown-it-py==1.1.0
mdformat==0.7.10
mdformat-gfm==0.3.3
mdformat_tables==0.4.1
mdit-py-plugins==0.2.8
tomli==1.2.2
uc-micro-py==1.0.1
zipp==3.6.0
```

**Comment by butler54** [comment]
Let me know if I should move this issue to the plugin repo, I put it here as most of the issues appeared to be centrally managed.

**Comment by hukkin** [comment]
Ugh, this is not good. Thanks a lot for the issue though!

I've moved to mdformat-gfm issue tracker https://github.com/hukkin/mdformat-gfm/issues/17

**Comment by hukkin** [comment]
This should be fixed in `mdformat-gfm` v0.3.4
